docs/kb/semgrep-ci/git-command-errors.md

```diff
@@ -0,0 +1,61 @@
+---
+tags:
+  - Git
+  - Semgrep in CI
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+# Failed to run a git command during a pull or merge request scan
+
+When running Semgrep in CI with a pull or merge request as the triggering event, Semgrep runs some additional `git` commands to determine the behavior for the scan. The scan exits with an error if these commands fail. A message like the following shows in the output:
+
+```
+[ERROR] Command failed with exit code: 128
+-----
+Command failed with output:
+fatal: Not a valid object name master
+
+
+Failed to run 'git <command-details>'. Possible reasons:
+
+- the git binary is not available
+- the current working directory is not a git repository
+- the current working directory is not marked as safe
+    (fix with `git config --global --add safe.directory $(pwd)`)
+
+Try running the command yourself to debug the issue.
+```
+
+In addition to the potential reasons included in the message, there are two other common reasons for git command failure:
+
+## Clone depth is too shallow
+
+If a shallow git clone of the repository is used to fetch code, and the branch to be scanned has had many commits added since it was branched off the base branch, Semgrep may not be able to identify the base branch commit to compare the current pull or merge request branch with. In this case, the command that failed is typically:
+
+<pre class="language-bash"><code>git merge-base --all <span className="placeholder">SHA</span> FETCH_HEAD</code></pre>
+
+Semgrep uses the merge-base command to compare the tip of the pull or merge request branch with the base branch and determine where it branched off, so that it can accurately scan for changes made in the merge request rather than in the base branch. 
+
+If there isn't enough history to identify the branch point, this comparison can fail. This is most common if the git clone performed is shallow by default, or if the depth has been set to a small value via command line argument `--depth` or environment variable `GIT_DEPTH`.
+
+To resolve this issue, increase the clone depth to a larger value. A value such as 20 or 50 is sufficient to capture the information needed for most merge-base calculations.
+
+For GitLab CI, see [Limit the number of changes fetched during clone](https://docs.gitlab.com/ee/ci/pipelines/settings.html#limit-the-number-of-changes-fetched-during-clone) for instructions on configuring this value.
+
```

Comment:
Might it help to add [SEMGREP_GHA_MIN_FETCH_DEPTH](https://semgrep.dev/docs/semgrep-ci/ci-environment-variables/#semgrep_gha_min_fetch_depth)? - I believe it is the GitHub counterpart.

Reply:
Weirdly, it's not - it's actually to avoid erroneous reports of findings not changed. We have automation that fetches more GitHub material as indeed to avoid the shallow clone problem. But you're right that I should probably make a mention of how this behaves in GHA and refer folks to that variable for what it is for, just in case. I was kinda waffling on this but having you ask confirms it. 👍🏻